homeassistant/components/upnp.py

```diff
@@ -90,10 +88,8 @@
                 service = device.find_first_service(IP_SERVICE)
             if _service['serviceType'] == CIC_SERVICE:
                 unit = config.get(CONF_UNITS)
-                discovery.load_platform(hass, 'sensor',
-                                        DOMAIN,
-                                        {'unit': unit},
-                                        config)
+                hass.async_add_job(discovery.async_load_platform(
+                    hass, 'sensor', DOMAIN, {'unit': unit}, config))
```

Comment:
```python
await discovery.async_load_platform(...)
```

?

Reply:
No. That's a circular dependency. You can't wait till it is loaded because the platform depends on the component being setup, which is waiting for the platform to be loaded, which is …